osidb/tests/test_flaw.py

```diff
@@ -1194,3 +1195,79 @@ def test_flaw_affects_rhscl_invalid_collection(
         affect = AffectFactory(ps_module=ps_module, ps_component=ps_component)
         if alerts:
             assert set(alerts).issubset(affect._alerts)
+
+    @pytest.mark.parametrize(
+        "impact,affectedness,resolution,prefix,should_raise",
+        [
+            (
+                Affect.AffectImpact.LOW,
+                Affect.AffectAffectedness.AFFECTED,
+                Affect.AffectResolution.WONTREPORT,
+                "service",
+                False,
+            ),
+            (
+                Affect.AffectImpact.MODERATE,
+                Affect.AffectAffectedness.AFFECTED,
+                Affect.AffectResolution.WONTREPORT,
+                "service",
+                False,
+            ),
+            (
+                Affect.AffectImpact.IMPORTANT,
+                Affect.AffectAffectedness.AFFECTED,
+                Affect.AffectResolution.WONTREPORT,
+                "service",
+                True,
+            ),
+            (
+                Affect.AffectImpact.LOW,
+                Affect.AffectAffectedness.AFFECTED,
+                Affect.AffectResolution.WONTREPORT,
+                "regular",
+                True,
+            ),
+            (
+                Affect.AffectImpact.LOW,
+                Affect.AffectAffectedness.AFFECTED,
+                Affect.AffectResolution.WONTREPORT,
+                "invalid",
+                True,
+            ),
+        ],
+    )
+    def test_validate_wontreport_products(
+        self, impact, affectedness, resolution, prefix, should_raise
+    ):
+        """
+        Tests that only products associated services, having a impact
+        of LOW or MODERATE, can be marked as AFFECTED by a WONTREPORT affect
+        """
+        SERVICES_PRODUCTS.append("service-product")
```

Comment:
Use monkeypatch [1] instead, if you don't there's a risk that tests defined after this one that depend on `SERVICES_PRODUCTS` either directly or indirectly will use it with `"service-product"` as a valid value which might not be what we want.

[1] https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html